command/plugin_runtime_register.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+var (
+	_ cli.Command             = (*PluginRuntimeRegisterCommand)(nil)
+	_ cli.CommandAutocomplete = (*PluginRuntimeRegisterCommand)(nil)
+)
+
+type PluginRuntimeRegisterCommand struct {
+	*BaseCommand
+
+	flagType         string
+	flagOCIRuntime   string
+	flagCgroupParent string
+	flagCPUNanos     int64
+	flagMemoryBytes  int64
+}
+
+func (c *PluginRuntimeRegisterCommand) Synopsis() string {
+	return "Registers a new plugin runtime in the catalog"
+}
+
+func (c *PluginRuntimeRegisterCommand) Help() string {
+	helpText := `
+Usage: vault plugin runtime register [options] NAME
+
+  Registers a new plugin runtime in the catalog. Currently, Vault only supports registering runtime of container type.
+The OCI runtime must be available on Vault's host. If no OCI runtime is specified, Vault will use \"runsc\", gVisor's OCI runtime.
+
+  Register the plugin runtime named my-custom-plugin-runtime:
+
+      $ vault plugin runtime register -type=container -oci_runtime=my-oci-runtime my-custom-plugin-runtime
+
+  Register a plugin runtime with custom arguments:
+
+      $ vault plugin runtime register \
+          -oci_runtime=my-oci-runtime \
+          -type=container my-custom-plugin
+
+` + c.Flags().Help()
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *PluginRuntimeRegisterCommand) Flags() *FlagSets {
+	set := c.flagSet(FlagSetHTTP)
+
+	f := set.NewFlagSet("Command Options")
+
+	f.StringVar(&StringVar{
+		Name:       "type",
+		Target:     &c.flagType,
+		Completion: complete.PredictAnything,
+		Usage:      "Plugin runtime type. Vault currently only supports \"container\" runtime type.",
+	})
+
+	f.StringVar(&StringVar{
+		Name:       "oci_runtime",
+		Target:     &c.flagOCIRuntime,
+		Completion: complete.PredictAnything,
+		Usage:      "OCI runtime. Default is \"runsc\", gVisor's OCI runtime.",
+	})
+
+	f.StringVar(&StringVar{
+		Name:       "cgroup_parent",
+		Target:     &c.flagCgroupParent,
+		Completion: complete.PredictAnything,
+		Usage:      "Parent cgroup for the container runtime",
```

Comment:
```suggestion
		Usage:      "Parent cgroup to set for each container. This can be used to control the total resource usage for a group of plugins.",
```